src/Features/Core/Portable/MoveStaticMembers/AbstractMoveStaticMembersRefactoringProvider.cs

```diff
@@ -38,26 +38,16 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 return;
             }
 
-            var selectedType = semanticModel.GetEnclosingNamedType(span.Start, cancellationToken);
-            if (selectedType == null)
+            var selectedMember = semanticModel.GetDeclaredSymbol(memberDeclaration, cancellationToken);
+            if (selectedMember is null || selectedMember.ContainingType is null ||
+                !selectedMember.IsStatic || !MemberAndDestinationValidator.IsMemberValid(selectedMember))
             {
                 return;
             }
 
-            var selectedMembers = selectedType.GetMembers()
-                .WhereAsArray(m => m.IsStatic &&
-                    MemberAndDestinationValidator.IsMemberValid(m) &&
-                    m.DeclaringSyntaxReferences.Any(static (sr, memberDeclaration) => memberDeclaration.FullSpan.Contains(sr.Span), memberDeclaration));
-            if (selectedMembers.IsEmpty)
-            {
-                return;
-            }
-
-            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
-
-            var action = new MoveStaticMembersWithDialogCodeAction(document, span, service, selectedType, context.Options, selectedMember: selectedMembers[0]);
+            var action = new MoveStaticMembersWithDialogCodeAction(document, memberDeclaration.Span, service, selectedMember.ContainingType, context.Options, selectedMember: selectedMember);
 
-            context.RegisterRefactoring(action, selectedMembers[0].DeclaringSyntaxReferences[0].Span);
```

Comment:
why this change?

Reply:
Ah, I incorrectly assumed that the overload without a provided span used the Span property on the provided `CodeAction`, but I see it fills in null if not provided. I'll change it to `memberDeclaration.Span`. I changed it initially because we want to make changes on the member we selected, not just the first reference to the first member of the type.

Reply:
More generally, I removed that code above (where we get all the members and validate each one) because if we are currently selecting a valid member for refactoring, then we have at least one member that is valid, so we don't need to check all the members.